### ISSUE TITLE
Ordering and visualization of All day events

### ISSUE THREAD

**Issue by yvolk** [andstatus/todoagenda#48]
### Current state (as of ToDo Agenda v.4.2.3) :
1. All day events (both Calendar events and tasks) are shown at the same time as events, scheduled at midnight of the same day.
2. No time is shown for All day events. Whereas e.g. "00:00" is shown for Calendar events, starting at midnight.
3. Actually, no time is shown for any Tasks, so All day tasks are visually indistinguishable from tasks, starting at midnight.

### Desired state
1. An option to choose All day events placement:
- At the top of the day's events (for today - after past events, if they are shown)
- At the bottom of the day's events (for today - before tasks without start and due dates, if they are shown)

2. Show time for tasks that do have time set AND that are shown in the list according to their date and time.
 This means that time is not shown for All day tasks, and e.g. for overdue tasks that are shown Today.

**Comment by Torenga** [comment]
I have some suggestions for Desired State 2., based on the existing icons and layout.

Suggestion for icon:

![suggested all day tasks](https://user-images.githubusercontent.com/68231509/92989943-3c4cc980-f4d8-11ea-83bd-3409f6ee84b8.png)

Suggestion for text:

`24h`
`All Day`
`Today`

---

But if Desired State 1. is implemented, the all day entries are already visually separated from events and tasks with a due time.

I'm not sure the widget need neither text or icons to show it's all day, as I assume events and tasks set at a date but not having a due time is all day by default? Example:

```
                         Monday 14 September
| Free entry at the museum all week!
14.sep - 19.sep
v Give feedback on #48
v Call Yuri when I find time
| Wash clothes and hang to dry
09:00-11:00
| Visit the museum
11:30-15.00
v Buy a poster from the exhibit
12:00
v Add money to travel card
15:00

                         Tuesday 15 September
| Free entry at the museum all week!
14.sep - 19.sep
v Fold dry clothes
| Meeting at work
10:00-12:00
v Check for replies to #48
18:00
```
But perhaps a clear separator would make it easier to distinguish between all-day and entries with due time:

```
                         Monday 14 September
| Free entry at the museum all week!
14.sep - 19.sep
v Give feedback on #48
v Call Yuri when I find time
AGENDA
| Wash clothes and hang to dry
09:00-11:00
| Visit the museum
11:30-15.00
v Buy a poster from the exhibit
12:00
v Add money to travel card
15:00
```

**Comment by yvolk** [comment]
Released to Beta channel, see #3

**Comment by Torenga** [comment]
> Released to Beta channel, see #3

It looks perfect. :heart: Thank you for considering my suggestions!

I will stay on the beta until 4.2.4 drops on F-Droid and report any bugs. Definitively a daily use widgetapp from now on.

**Comment by yvolk** [comment]
Thank you, @Torenga. Together we are designing far better app, really.
